STM32WB update drivers version to CUBE V1.8.0

jeromecoutant opened the pull request:
### Summary of changes <!-- Required -->

ST Cube drivers version is updated from V1.7.0 to V1.8.0
which is the latest official version from:

https://github.com/STMicroelectronics/STM32CubeWB/

@ARMmbed/team-st-mcd


#### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->

<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

Verified with ST CI

    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

jeromecoutant committed:
STM32WB: STM32Cube_FW_WB_V1.8.0

https://github.com/STMicroelectronics/STM32CubeWB

0xc0170 commented on connectivity/drivers/ble/FEATURE_BLE/TARGET_STM32WB/STM32Cube_FW/ble_bufsize.h:
```diff
@@ -0,0 +1,161 @@
+ *****************************************************************************
+  * @attention
+  *
+  * <h2><center>&copy; Copyright (c) 2019 STMicroelectronics.
```
I recall seeing other cube drivers using SPDX, not this version?

jeromecoutant replied:
Mmm, this is because file comes from CubeFW without any modification....

0xc0170 replied:
Can you add SPDX identifiers and send also upstream?

jeromecoutant replied:
As I said, file comes from https://github.com/STMicroelectronics/STM32CubeWB/blob/master/Middlewares/ST/STM32_WPAN/ble/core/ble_bufsize.h
Like in https://github.com/ARMmbed/mbed-os/tree/master/targets/TARGET_STM/TARGET_STM32WB/STM32Cube_FW
I prefer to avoid any modification to ease driver updates when there is a new ST package

0xc0170 replied:
Travis fails as SPDX is required for all new files :(

the only way to address this would be to add the driver to license check ignore.

1. add to ignore for now,
2. add SPDX to the usptream repo (link above)
3. update drivers in this tree once the update is due

what do you think?

jeromecoutant replied:
You should ignore all files `*/STM32Cube_FW/*`
in `targets/TARGET_STM`
and in other directories like this one....

0xc0170 replied:
Ok, please add them to codecheckignore list. This will make Travis green.

0xc0170 committed:
travis: ignore spdx reports

Do not force red status for PRs due to SPDX missing (license must be there). We will
add better SPDX checks later (more work required to enable these for 3rd party code).
